src/Features/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,403 @@
+            else if (IsAt("\r") || IsAt("\n"))
+            {
+                Position++;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetCharsToCurrentPosition(start));
+            }
+
+            return null;
+        }
+
+        public JsonTrivia? ScanComment()
+        {
+            if (IsAt("//"))
+            {
+                return ScanSingleLineComment();
+            }
+            else if (IsAt("/*"))
+            {
+                return ScanMultiLineComment();
+            }
+            else if (IsAt("/"))
+            {
+                var start = Position;
+                Position++;
+
+                var chars = GetCharsToCurrentPosition(start);
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars,
+                    ImmutableArray.Create(new EmbeddedDiagnostic(
+                        FeaturesResources.Error_parsing_comment,
+                        GetSpan(chars))));
+            }
+
+            return null;
+        }
+
+        private JsonTrivia ScanSingleLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length &&
+                   this.CurrentChar is var ch &&
+                   ch != '\r' && ch != '\n')
+            {
+                Position++;
+            }
+
+            var chars = GetCharsToCurrentPosition(start);
+            if (Position == start + 2)
+            {
+                // Note: json.net reports an error if the file ends with "//", so we just
+                // preserve that behavior.
+                var diagnostics = ImmutableArray.Create(new EmbeddedDiagnostic(
+                    FeaturesResources.Unterminated_comment,
+                    GetSpan(chars)));
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars, diagnostics);
+            }
+
+            return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars);
+        }
+
+        private JsonTrivia ScanMultiLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length &&
+                   !IsAt("*/"))
+            {
+                Position++;
+            }
+
+            if (IsAt("*/"))
+            {
+                Position += 2;
+                return CreateTrivia(JsonKind.MultiLineCommentTrivia, GetCharsToCurrentPosition(start));
+            }
+
+            Debug.Assert(Position == Text.Length);
+            return CreateTrivia(JsonKind.MultiLineCommentTrivia, GetCharsToCurrentPosition(start),
+                ImmutableArray.Create(new EmbeddedDiagnostic(
+                    FeaturesResources.Unterminated_comment,
+                    GetTextSpan(start, Position))));
+        }
+
+        private TextSpan GetTextSpan(int startInclusive, int endExclusive)
+            => TextSpan.FromBounds(Text[startInclusive].Span.Start, Text[endExclusive - 1].Span.End);
+
+        private bool IsAt(string val)
```

Comment:
```suggestion
        private bool IsTextAtCurrent(string val)
```

Reply:
I'm fine with IsAt.  It also matches the regex lexer :)